vpr/src/util/vpr_utils.h

```diff
@@ -146,6 +148,8 @@ int get_max_num_pins(t_logical_block_type_ptr logical_block);
 
 bool is_tile_compatible(t_physical_tile_type_ptr physical_tile, t_logical_block_type_ptr logical_block);
 
+bool is_tile_compatible(t_physical_tile_type_ptr physical_tile, t_logical_block_type_ptr logical_block, t_pl_loc loc);
```

Comment:
I expect the t_pl_loc is needed to see which capacity location the logical block is in.

If that is the case, we probably don't need the physical_tile as well (as it is also determined by the loc).

It may make sense to factor this out so there is a:
```cpp
bool is_subtile_compatible(t_sub_tile sub_tile, t_logical_block_type_ptr logical_block);
```
which is_tile_compatible() calls internally.

Thoughts?

Reply:
This function was actually thought to be used in cases where the placement is not yet completed, to check whether specific locations were compatible (e.g. read_place.cpp, move_utils.cpp).

It works the other way around, meaning that, first, there is a check to test whether a specific sub tile index is compatible within a physical tile, and then, the logic block itself is checked as well, to see if that is compatible with the physical tile (by calling the original `is_tile_compatible` function.